pkg/apis/configuration/validation/transportserver_test.go

```diff
@@ -387,6 +387,79 @@ func TestValidateListenerProtocol(t *testing.T) {
 	}
 }
 
+func TestValidateTSUpstreamHealthChecks(t *testing.T) {
+	tests := []struct {
+		healthCheck *v1alpha1.HealthCheck
+		msg         string
+	}{
+		{
+			healthCheck: nil,
+			msg:         "nil health check",
+		},
+		{
+			healthCheck: &v1alpha1.HealthCheck{},
+			msg:         "non nil health check",
+		},
+		{
+			healthCheck: &v1alpha1.HealthCheck{
+				Enabled:  true,
+				Timeout:  "30s",
+				Jitter:   "5s",
+				Port:     88,
+				Interval: "10",
+				Passes:   3,
+				Fails:    4,
+			},
+			msg: "valid Health check",
+		},
+	}
+	for _, test := range tests {
+		allErrs := validateTSUpstreamHealthChecks(test.healthCheck, field.NewPath("healthCheck"))
+		if len(allErrs) > 0 {
+			t.Errorf("validateUpstreamHealthChecks() returned errors %v  for valid input for the case of %s", allErrs, test.msg)
+		}
+	}
+}
+
+func TestValidateTSUpstreamHealthChecksFails(t *testing.T) {
+	tests := []struct {
+		healthCheck *v1alpha1.HealthCheck
+		msg         string
+	}{
+		{
+			healthCheck: &v1alpha1.HealthCheck{
+				Enabled:  true,
+				Timeout:  "-30s",
+				Jitter:   "5s",
+				Port:     88,
+				Interval: "10",
+				Passes:   3,
+				Fails:    4,
+			},
+			msg: "invalid timeout",
```

Comment:
I just thought that the other fields using the same validation methods as those 2, either validating a string or validating an int that it would be redundant to test for all the fields, no?

Reply:
in this case, we're testing `validateTSUpstreamHealthChecks`,  not `validateTime` and `validatePositiveIntOrZero`. Essentially, our cases here  test that validateTSUpstreamHealthChecks propagates errors from `validateTime` and `validatePositiveIntOrZero`, not that `validateTime` and `validatePositiveIntOrZero` work correctly - that is tested in the tests for those functions.